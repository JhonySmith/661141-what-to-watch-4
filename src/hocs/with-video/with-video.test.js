import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import withVideo from "./with-video.js";

const movie =
{
  title: `Fantastic Beasts: The Crimes of Grindelwald`,
  image: `https://upload.wikimedia.org/wikipedia/commons/f/f6/70mm_film_print_with_DTS_sound.jpg`,
  preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  genre: `horror`,
};

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWraped = withVideo(MockComponent);

describe(`With-video snap tests`, () => {
  it(`With video hoc render correctly`, () => {
    const tree = renderer.create(
        <MockComponentWraped
          muted={true}
          movie={movie}
        />, {
          createNodeMock: () => {
            return {};
          }
        }).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
