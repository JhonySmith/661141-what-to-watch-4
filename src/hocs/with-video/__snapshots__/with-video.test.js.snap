// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With-video snap tests With video hoc render correctly 1`] = `
<div>
  <video
    height="175"
    poster="https://upload.wikimedia.org/wikipedia/commons/f/f6/70mm_film_print_with_DTS_sound.jpg"
    type="video/mp4"
    width="280"
  >
    Sorry, your browser doesnt support embedded videos.
  </video>
</div>
`;
